File: Backend/src/main/java/com/example/demo/Models/TypeModel.java
package com.example.demo.Models;

import com.example.demo.DemoApplication;
import com.fasterxml.jackson.core.JsonProcessingException;

public class TypeModel {

    private Long id;
    private String type;

    public TypeModel() {

    }

    public TypeModel(Long id, String type) {
        this.id = id;
        this.type = type;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        try {
            return DemoApplication.objectMapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: Frontend/lib/Model/TypeModel.dart
class TypeModel {
  String? id;
  String? type;

  TypeModel({this.id, this.type});

  TypeModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    type = json["type"];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['type'] = type;
    return data;
  }

  get getId => id;

  get getType => type;
}
